### PR TITLE
[ROCM] update fluid memory for rocm35 (part1), test=develop

### PR DESCRIPTION
**Pull request by qili93**
### PR types

New features

### PR changes

Others

### Describe

[ROCM] update fluid memory for rocm35 (part1)

### CODE REVIEW

File: paddle/fluid/memory/allocation/allocator_facade.cc [hunk 1]
```diff
@@ -70,7 +70,7 @@ class AllocatorFacadePrivate {
           InitNaiveBestFitXPUAllocator(platform::XPUPlace(dev_id));
         }
 #endif
-#ifdef PADDLE_WITH_CUDA
+#if defined(PADDLE_WITH_CUDA) || defined(PADDLE_WITH_HIP)
```
Comment (zhiqiu) [inline paddle/fluid/memory/allocation/allocator_facade.cc:73]: Maybe we can create a new MACRO, eg,
`PADDLE_WITH_GPU  = PADDLE_WITH_CUDA|| PADDLE_WITH_HIP`

File: paddle/fluid/memory/allocation/cuda_device_context_allocator.h [hunk 2]
```diff
@@ -102,19 +109,24 @@ class CUDADeviceContextAllocator : public Allocator {
     platform::CUDADeviceGuard guard(place_.device);
     auto allocation =
         new CUDADeviceContextAllocation(memory::Alloc(place_, size));
-    // Wait for the event on stream
+// Wait for the event on stream
+#ifdef PADDLE_WITH_HIP
+    PADDLE_ENFORCE_CUDA_SUCCESS(hipEventRecord(event_, default_stream_));
```
Comment (zhiqiu) [inline paddle/fluid/memory/allocation/cuda_device_context_allocator.h:114]: Better add hip support for PADDLE_ENFORCE_CUDA_SUCCESS to generate better error message.

Comment (qili93) [reply paddle/fluid/memory/allocation/cuda_device_context_allocator.h:114]: will address in enforce.h later as discussed in PR#30758

File: paddle/fluid/memory/allocation/allocator_facade.cc (hunk 1 above)
Comment (qili93) [reply paddle/fluid/memory/allocation/allocator_facade.cc:73]: Keep PADDLE_WITH_HIP as cudnn vs miopen has big difference.

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**